src/engine/GameObject.cpp

```diff
@@ -1,13 +1,18 @@
 #include "GameObject.hpp"
 
+using namespace admirals;
 using namespace admirals::scene;
 
-GameObject::GameObject(const Vector3 &pos) : m_position(pos) {}
+GameObject::GameObject(const std::string &name, const Vector3 &pos)
+    : m_name(name), m_position(pos) {}
+
+std::string GameObject::name() const { return m_name; }
+float GameObject::order() const { return m_position.z(); }
```

Comment:
This is unclear for the user I believe. Settings the order should be explicit in the constructor, not done implicitly by "z"-part of the position.